src/prefect/cli/heartbeat.py

```diff
@@ -94,6 +95,15 @@ def flow_run(id, num):
     logger = get_logger("heartbeat")
     iter_count = 0
 
+    # Lower the OOM score for the hearbeat process so it is not killed before the flow
+    # run
+    try:
+        if sys.platform == "linux":
+            with open("/proc/self/oom_score_adj", "wb") as f:
+                f.write("0".encode("ascii"))
```

Comment:
The score ranges from -1000 to 1000. I initially set it to -500, but a brief internet search makes it look like setting the score below 0 requires admin permissions but I kind of assume it starts at 0? That means that to make this work we may need to set the flow run process `prefect execute flow-run` to a higher value e.g. `100` then set this one back to `0` since it would inherit the parent's setting? Hmm. Not as simple as I hoped.

Reply:
See a decent description at https://askubuntu.com/a/1349313